hcl/hcl.go

```diff
@@ -146,6 +155,31 @@ type TerramateParser struct {
 	Blocks ast.Blocks
 }
 
+var stackIDRegex = regexp.MustCompile("^[a-zA-Z0-9_-]{1,64}$")
+
+// NewStackID creates a new StackID with the given string as its id.
+// It guarantees that the id passed is a valid StackID value,
+// an error is returned otherwise.
+func NewStackID(id string) (StackID, error) {
+	if !stackIDRegex.MatchString(id) {
+		return StackID{}, errors.E("Stack ID %q doesn't match %q", id, stackIDRegex)
+	}
+	return StackID{id: &id}, nil
+}
+
+// Value returns the ID string value and true if this StackID is defined,
+// it returns "" and false otherwise.
+func (s StackID) Value() (string, bool) {
+	// This is like a poor's man option type.
```

Comment:
I dont think the comments are needed because it looks clear as `id` is a pointer. But it's your call, you can keep it.

Reply:
ah we can remove, not in love with the comment :laughing: